### PR TITLE
Changes to allow OAuth integration with nDelius users

### PR DESCRIPTION
**Pull request by awhodgson256**
Initial implementation of nDelius user oAuth for review

### CODE REVIEW

File: mockApis/interventionsService.ts [hunk 1]
```diff
@@ -0,0 +1,1007 @@
+import Wiremock from './wiremock'
+
+export default class InterventionsServiceMocks {
```
Comment (rgforsyth) [inline mockApis/interventionsService.ts:3]: I assume this was just copied temporarily to get things working or as an example for when we need to mock things later.

File: mockApis/referAndMonitorAndDelius.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import Wiremock from './wiremock'
+
+export default class ReferAndMonitorAndDeliusMocks {
```
Comment (rgforsyth) [inline mockApis/referAndMonitorAndDelius.ts:3]: As above

File: server/utils/errorMessages.ts [hunk 3]
```diff
@@ -0,0 +1,384 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+// Explicit return types would make this file very messy and not much more
+// informational — everything returned is a string, since they’re messages!
+
+const errorHandlerAccessErrorMessages = {
```
Comment (rgforsyth) [inline server/utils/errorMessages.ts:5]: Again, I assume most of these will be cleared later on

File: mockApis/auth.ts [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+  private readonly tokenVerification = new TokenVerifiationMocks(this.wiremock)
+
+  private createToken = (authSource: string): string => {
+    const payload = {
+      user_name: 'USER1',
+      scope: ['read'],
+      auth_source: authSource,
+      authorities: authSource === 'auth' ? ['ROLE_CRS_PROVIDER'] : ['ROLE_PROBATION'],
+      jti: '83b50a10-cca6-41db-985f-e87efb303ddb',
+      client_id: 'interventions',
```
[the remark this answers is not in the document]
Comment (awhodgson256) [reply mockApis/auth.ts:18]: @rgforsyth Id's and secrets currently taken from R&M, I'd like feedback on where we are supposed to source / generate these

[the remark this answers is not in the document]
Comment (amolphalnikar) [reply mockApis/auth.ts:18]: We need to remove ROLE_CRS_PROVIDER as it is for Service providers. Our service will use ROLE_PROBATION as a default role for all Probation Practitioners who log in.

**Comment by amolphalnikar** [comment]
Hi @awhodgson256, I feel, we should keep only minimal things to get the auth working. Any other things can be added later on. This will avoid any confusion and when we request a review from a lead dev, there won't be any unnecessary comments and our initial skeleton project structure would be in good shape.

**Review by amolphalnikar — APPROVED** [review]

**Pull request merged**